content/learn/book/your-first-game/_index.md

```diff
@@ -0,0 +1,685 @@
+    delta_seconds: f32,
+    seconds_since_startup: f64,
+    startup: Instant,
+}
+```
+Inside the function, that is called every frame, we iterate over all `Components` in our scene that match the type Paddle.  
+Then we check if the left or right cursor key is pressed and we set the direction variable accordingly.  
+Finally we access the translation (a fancy word for position) field of the `Transform` instance and use it to change the horizontal (x) paddle position on the playing field.  
+As done here, it is good practice to factor in time.delta_seconds(). This makes sure that the behavior of the paddle is independent of the frame rate.
+We also have to make sure that the paddle doesn't move off of the screen. So we limit the paddle inside of 380 pixels to the right and left of the center position.
+
+## Next let's add some walls.
+
+```rs
+    let wall_material = materials.add(Color::rgb(0.8, 0.8, 0.8).into());
+    let wall_thickness = 10.0;
+    let bounds = Vec2::new(900.0, 600.0);
+
+    commands
+        // left
+        .spawn(SpriteBundle {
+            material: wall_material.clone(),
+            transform: Transform::from_xyz(-bounds.x / 2.0, 0.0, 0.0),
+            sprite: Sprite::new(Vec2::new(wall_thickness, bounds.y + wall_thickness)),
+            ..Default::default()
+        })
+        .with(Collider::Solid)
+        // right
+        .spawn(SpriteBundle {
+            material: wall_material.clone(),
+            transform: Transform::from_xyz(bounds.x / 2.0, 0.0, 0.0),
+            sprite: Sprite::new(Vec2::new(wall_thickness, bounds.y + wall_thickness)),
+            ..Default::default()
+        })
+        .with(Collider::Solid)
+        // bottom
+        .spawn(SpriteBundle {
+            material: wall_material.clone(),
+            transform: Transform::from_xyz(0.0, -bounds.y / 2.0, 0.0),
+            sprite: Sprite::new(Vec2::new(bounds.x + wall_thickness, wall_thickness)),
+            ..Default::default()
+        })
+        .with(Collider::Solid)
+        // top
+        .spawn(SpriteBundle {
+            material: wall_material,
+            transform: Transform::from_xyz(0.0, bounds.y / 2.0, 0.0),
+            sprite: Sprite::new(Vec2::new(bounds.x + wall_thickness, wall_thickness)),
+            ..Default::default()
+        })
+        .with(Collider::Solid);
+```
+Add this code to the inside of the startup function, just before it ends and run the program with `cargo run`.
+You should now see walls inside of your game just like this:
+
+![](breakout3.png)
+
+With all that we have learned so far, it should be pretty obvious what the code above does. But let's go over some interesting details.
+You might have noticed that except for the last wall, every other wall has received a clone of the wall_material.
+Why is that necessary?  
+Well two fold: It is usually good practice to give every object in a game its own material. Otherwise it can happen, that you change the color on one object and suddenly you
```

Comment:
```suggestion
Well, the reason is twofold: it is usually good practice to give every object in a game its own material. Otherwise it can happen, that you change the color on one object and suddenly you
```